modules/MinimumSpanningTree/src/main/java/com/carlschroedl/gephi/plugin/minimumspanningtree/KruskalsAlgorithm.java

```diff
@@ -0,0 +1,409 @@
+             * spanning tree. In this way the algorithm will always terminate at
+             * or before 100%.
+             */
+            Progress.start(progressTicket, graph.getEdgeCount());
+
+            //Convert all edges to the Comparable KEdge, then add to PriorityQueue
+            for (Edge e : graph.getEdges()) {
+                edgeQ.add(new KEdge(e));
+            }//endforeach
+
+            //By definition, a tree of n node must have n-1 edges
+            int finishSize = graph.getNodeCount() - 1;
+
+            KEdge tempKEdge; //KEdge popped off the PriorityQueue
+
+            while (edgesInST < finishSize) {
+                tempKEdge = edgeQ.remove();
+                if (tempKEdge.getSource().getDjsPointer().find()
+                        != tempKEdge.getTarget().getDjsPointer().find()) {
+                    //and yes, it IS ok to use != instead of !blah.equals()
+
+                    //change edge's Spanning Tree Attribute to non-default value
+                    tempKEdge.edge.setAttribute(SPANNING_TREE_COLUMN_ID, IN_SPANNING_TREE);
+
+                    ++edgesInST;
+                    this.STweight += tempKEdge.weight;
+
+
+                    Unionizer unionizer = new Unionizer();
+                    unionizer.union(
+                            tempKEdge.getSource().getDjsPointer(),
+                            tempKEdge.getTarget().getDjsPointer());
+                }
+
+                Progress.progress(progressTicket, edgesInST);
+                if (cancel) {
+                    //remove the spanning tree column
+                    graphModel.getEdgeTable().removeColumn(stEdgeCol);
+                    break;
+                }
+            }//end while
+
+        } catch (Exception e) {
+            e.printStackTrace();
+        } finally {
+            //Unlock graph
+            graph.writeUnlock();
+        }
+    }
+
+    @Override
+    public JPanel getOptions() {
+        throw new UnsupportedOperationException("Not supported yet.");
+    }
+
+    @Override
+    public String getName() {
+        return name;
+    }
+
+    @Override
+    public boolean cancel() {
+        this.cancel = true;
+        return true;
+    }
+
+    @Override
+    public void setProgressTicket(ProgressTicket progressTicket) {
+        this.progressTicket = progressTicket;
+    }
+
+    @Override
+    public String getReport() {
+        return "Number of edges in the minimum spanning tree: "
+                + this.edgesInST
+                + "<br/> Weight of the minimum spanning tree: "
+                + this.STweight
+                + "<br/> Every edge in the minimum spanning tree now has an attribute \"" + ST_COL_NAME + "\" set equal to \"1\"."
+                + "<br/> The minimum spanning tree can be visualized by creating a Filter on the Attribute \"" + ST_COL_NAME + "\" with value = \"1\"";
+    }
+
+    // <editor-fold defaultstate="collapsed" desc="Inner Helper Classes">
```

Comment:
When I was a student I made the unfortunate decision to write many of the following classes as inner classes. I hope to convert these into standalone public classes in a subsequent refactor (https://github.com/carlschroedl/gephi-plugins/issues/4).